packages/components/src/Dropdown/Dropdown.tsx

```diff
@@ -48,6 +55,23 @@ type DropdownProps = ListboxProps & {
    * Optional className for additional styling.
    */
   className?: string;
+  /**
+   * Render dropdown button that is only as wide as the content.
+   *
+   * When defining compact dropdown need to provide optionsWidth to
+   * define width of options menu, and optionally provide optionsAlign
+   * if desired to right align to dropdow button.
+   */
+  compact?: boolean;
```

Comment:
_Should we make this a variant="compact" prop instead?_, I wondered to myself. Right now I think since this is only toggling sizing, it's fine as-is -- but if the "compact" style also had e.g. a background color change, then the `variant` prop could make sense

Reply:
`size?: "compact" | "full"` could make sense though 🤔

Reply:
I've landed with `variant` because it is unclear by size if that meant the button or the whole component.